### PR TITLE
[#49] Support for CSV content type

### PR DESCRIPTION
**Pull request by pkdash**

### CODE REVIEW

File: hsmodels/schemas/aggregations.py [hunk 1]
```diff
@@ -233,7 +234,7 @@ class MultidimensionalMetadataIn(BaseAggregationMetadataIn):
         title="Spatial coverage",
         description="An object containing the geospatial coverage for the aggregation expressed as either a bounding box or point",
     )
-    spatial_reference: Union[MultidimensionalBoxSpatialReference, MultidimensionalPointSpatialReference] = Field(
+    spatial_reference: MultidimensionalBoxSpatialReference = Field(
```
Comment (devincowan) [inline hsmodels/schemas/aggregations.py:237]: I'm not seeing why this change?

File: hsmodels/schemas/aggregations.py [hunk 2]
```diff
@@ -561,3 +562,40 @@ class ModelInstanceMetadata(ModelInstanceMetadataIn):
...
+        description="A string expressing the aggregation type from the list of HydroShare aggregation types",
+        json_schema_extra={"readOnly": True},
+    )
+
+    url: AnyUrl = Field(
+        title="Aggregation URL", description="An object containing the URL of the aggregation", frozen=True,
+        json_schema_extra={"readOnly": True},
+    )
+
+    rights: Optional[Rights] = Field(
```
Comment (devincowan) [inline hsmodels/schemas/aggregations.py:594]: should there be any additional fields besides `url` and `rights` in this class?

**Review by devincowan — APPROVED** [review]

File: hsmodels/schemas/aggregations.py (hunk 1 above)
Comment (pkdash) [reply hsmodels/schemas/aggregations.py:237]: When testing I was getting errors with one specific netcdf file. Initially I thought the use of `MultidimensionalPointSpatialReference` might be causing the errors since hydroshare doesn't support point type spatial coverage for netcdf. You can see here (https://github.com/hydroshare/hydroshare/blob/develop/hs_file_types/models/netcdf.py#L96) when we serialize the OriginalCoverage (same as spatial reference) model to xml, we only use the coverage type to be Box. That's the reason I removed it. However, the errors that I was getting with one netcdf file was not due to the `MultidimensionalPointSpatialReference`. I ran all the unit tests after this change and they all passed. 

It seems if I keep this change, then I need to remove the class definition `MultidimensionalPointSpatialReference` as well as any references to it.

Comment (devincowan) [reply hsmodels/schemas/aggregations.py:237]: okay

Comment (pkdash) [reply hsmodels/schemas/aggregations.py:236]: I have now cleaned up code related to `MultidimensionalPointSpatialReference`.

**Review by devincowan — APPROVED** [review]
